R/docvars.R

```diff
@@ -239,19 +239,31 @@ set_docvars <- function(dvars, field = NULL, value)  {
     } else {
         dvars_system <- select_fields(dvars, c("text", "system"))
         dvars_user <- select_fields(dvars, "user")
+        
         if (is.dfm(value))
             value <- convert(value, to = "data.frame")[, -1, drop = FALSE]
+        
         if (is.null(field)) {
-            if (is.data.frame(value)) {
-                if (nrow(value) != nrow(dvars))
-                    stop(message_error("docvar_mismatch"))
-                result <- cbind(dvars_system, value)
-            } else {
+            if (is.null(value)) {
                 result <- dvars_system
+            } else {
+                if (is.matrix(value))
+                    value <- as.data.frame(value, stringsAsFactors = FALSE)
+                if (is.data.frame(value)) {
+                    if (nrow(value) != nrow(dvars))
+                        stop(message_error("docvar_mismatch"), call. = FALSE)
+                    if (any(missingnames <- nzchar(names(value), keepNA = TRUE))) {
```

Comment:
I prefer to raise error for data.frame without names. Why do we need to auto-generate names?

Reply:
I would be better to drop all the unnamed or duplicated columns than to automatically renaming.

Reply:
I'm ok with raising an error, but 
1.  This is the current behaviour, and
2. It's more robust for users, especially those who assign a matrix as value. (Note that the coercion from matrix format still takes place, but only if `value` is input as a matrix.)